patch_test.go

```diff
@@ -65,14 +65,40 @@ var _ = gc.Suite(&PatchEnvironmentSuite{})
 func (*PatchEnvironmentSuite) TestPatchEnvironment(c *gc.C) {
 	const envName = "TESTING_PATCH_ENVIRONMENT"
 	// remember the old value, and set it to something we can check
-	oldValue := os.Getenv(envName)
+	oldValue, oldValueSet := os.LookupEnv(envName)
+	defer func() {
+		if oldValueSet {
+			os.Setenv(envName, oldValue)
```

Comment:
Use `_ = os.Setenv(...)` to make it glaringly obvious that you're ignoring an error, here and elsewhere.